src/app/pages/single-recipe/single-recipe.component.ts

```diff
@@ -1,56 +1,142 @@
-import { faStar } from '@fortawesome/free-regular-svg-icons';
-import { faStar as faStarSolid , faPlusMinus } from '@fortawesome/free-solid-svg-icons';
+import { faStar, faTrashCan } from '@fortawesome/free-regular-svg-icons';
+import { faStar as faStarSolid, faPlusMinus, faStarHalfAlt } from '@fortawesome/free-solid-svg-icons';
 import { UsersService } from 'src/app/services/users/users.service';
 import { Recipe } from 'src/app/models/modelRecipe/recipe.model';
+import { DeleteRecipeService } from '../../services/delete/delete-recipe.service';
+import {WarningModalComponent} from "../../component/warning-modal/warning-modal.component";
+import {SuccessModalComponent} from "../../component/success-modal/success-modal.component";
 
 @Component({
   selector: 'app-single-recipe',
   templateUrl: './single-recipe.component.html',
-  styleUrls: ['./single-recipe.component.scss']
+  styleUrls: ['./single-recipe.component.scss'],
 })
 export class SingleRecipeComponent implements OnInit {
   recipe!: Recipe;
   routeSubscription!: Subscription;
-  commentText: string = '';
+  commentText = '';
   isLoading = true;
+  isUserCreateRecipe!: boolean;
+  rating = 0;
+  tempRating = 0;
+  isCommentTooShort = false;
 
-  constructor(private recipesService: RecipesService,
-    private route: ActivatedRoute ,
-    public converter : ConverterRecipesService,
-    public userService :UsersService)  {}
+  constructor(
+    private recipesService: RecipesService,
+    private route: ActivatedRoute,
+    public converter: ConverterRecipesService,
+    public userService: UsersService,
+    public deleteService: DeleteRecipeService,
+    private router: Router,
+    private dialog: MatDialog
+  ) {}
 
   ngOnInit(): void {
     this.routeSubscription = this.route.paramMap.subscribe((params: ParamMap) => {
-      const recipeName = params.get('name');
-      if (recipeName) {
-        this.recipesService.getRecipeByName(recipeName).subscribe((recipe: Recipe) => {
+      const recipeSlug = params.get('name');
+      if (recipeSlug) {
+        this.recipesService.getRecipeBySlug(recipeSlug).subscribe((recipe: Recipe) => {
           this.recipe = recipe;
           console.log(this.recipe);
           this.isLoading = false;
-
-        })
+          this.isUserCreateRecipe = this.userService.isCreateRecipe(this.recipe.id);
+        });
       }
     });
   }
 
+  setTempRating(rating: number, event: any): void {
+    const rect = (event.target as HTMLElement).getBoundingClientRect();
+    const x = event.clientX - rect.left; // x position within the element
+    const width = rect.right - rect.left;
+    this.tempRating = x > width / 2 ? rating : rating - 0.5;
+  }
+
+  resetTempRating(): void {
+    this.tempRating = this.rating;
+  }
+
+  setRating(rating: number, event: any): void {
+    const rect = (event.target as HTMLElement).getBoundingClientRect();
+    const x = event.clientX - rect.left; // x position within the element
```

Comment:
dead code or informative purpose ? delete if unnecessary

Reply:
i'm using for calculate position star for animation